socket.go

```diff
@@ -0,0 +1,38 @@
+package shuttle
+
+import (
+	"log"
+	"net"
+	"time"
+)
+
+type Socket struct {
+	net.Listener
+	Shuttle *Shuttle
+}
+
+func NewSocket(s *Shuttle) *Socket {
+	l, err := net.Listen("unix", s.config.Socket)
+	if err != nil {
+		log.Fatalln(err)
+	}
+
+	return &Socket{
+		Listener: l,
+		Shuttle:  s,
+	}
+}
+
+func (k *Socket) Start() {
+	for {
+		c, err := k.Accept()
+		if err != nil {
+			if err, ok := err.(*net.OpError); ok && err.Timeout() {
+				continue
+			}
+			return
+		}
+		c.SetDeadline(time.Now().Add(1 * time.Second))
```

Comment:
Also ReadLogLines uses buffered input, so it doesn't have direct control over how much is read from the socket at a time. w/o a deadline set the buffering in the reader could effectively wait forever for the last byte (depending on the length of the input vs. the length of the read buffer) and never process data.